agent/lib/kontena/load_balancers/configurer.rb

```diff
@@ -77,8 +77,11 @@ def ensure_config(container)
     # @param [Docker::Container] container
     def remove_config(container)
       name = container.labels['io.kontena.load_balancer.name']
-      service_name = container.service_name_for_lb
+      service_name = container.env_hash['KONTENA_LB_SERVICE_ALIAS'] || container.service_name_for_lb
+
       mode = container.env_hash['KONTENA_LB_MODE'] || 'http'
+      return if count_upstreams(name, service_name, mode) > 0
```

Comment:
The `lb:remove_config` -> `remove_config` only happens when terminating the `instance_number == 1` container. Are all the other upstreams guaranteed to be gone at that point?

i.e. do we terminate service instances in reverse-order, and wait for each service instance to be terminated before terminating the first instance?